packages/govern-subgraph/schema.graphql

```diff
@@ -1,22 +1,55 @@
-type Govern @entity {
+type ERC3000Registry @entity {
   id: ID!
   address: Bytes!
+  count: Int!
+  entries: [Entry!]
+}
+
+type Entry @entity {
+  id: ID!
   name: String!
-  queue: OptimisticQueue! @derivedFrom(field: "govern")
-  roles: [Role]!
+  dao: Govern!
+  queue: OptimisticQueue!
+  metadata: Bytes
+}
+
+type Govern @entity {
+  id: ID!
+  address: Bytes!
+  executions: [Execution!]
+  roles: [Role!]
+}
+
+type Execution @entity {
+  id: ID!
+  sender: Bytes!
+  actions: [Action!] @derivedFrom(field: "execution")
+  results: [Bytes!]
+}
+
+type Action @entity {
+  id: ID!
+  to: Bytes!
+  value: BigInt!
+  data: Bytes!
+  item: Item!
+  execution: Execution
 }
 
 type OptimisticQueue @entity {
```

Comment:
The `OptimisticQueue` change as well. I tried to decouple of the container concept and instead provide a more friendly and reach structure. Now we have the `items` on the queue (that are schedule) and then we also have `Challenge` items and `Veto` items.

I'm not convinced yet if we should also have the `Execution` here and how maybe a better way to link them with the `Govern` executions. This is one of the section that I'm unsure about the smartcontrats.

Reply:
I _definitely_ think executions should be here. The Queue's the main point of interaction with the DAO—we schedule, challenge and execute actions from it. If anything, we might just want to rethink how to link a Govern entity with its Queue entity to not have duplicate executions. 

Apart from this, love this change!